tests/integration/integration_suite_test.go

```diff
@@ -1,13 +1,316 @@
-package integration_test
+package integration
 
 import (
+	"log"
+	"strings"
+
+	. "github.com/k8ssandra/k8ssandra/tests/integration/steps"
+
+	"fmt"
+	"os"
 	"testing"
+)
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+const (
+	medusaTestTable    = "medusa_test"
+	medusaTestKeyspace = "medusa"
+	traefikNamespace   = "traefik"
 )
 
-func TestIntegration(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "Integration Suite")
+func TestMain(m *testing.M) {
+	os.Exit(m.Run())
+}
+
+func initializeCluster(t *testing.T) string {
+	log.Println(Step("Initializing cluster"))
+	CheckK8sClusterIsReachable(t)
+	InstallTraefik(t)
+	namespace := CreateNamespace(t)
+	CheckNamespaceWasCreated(t, namespace)
+	return namespace
+}
+
+func cleanupCluster(t *testing.T, namespace string) {
+	log.Println(Step("Cleaning up cluster"))
+	UninstallK8ssandraHelmRelease(t, namespace)
+	UninstallTraefikHelmRelease(t, traefikNamespace)
+	DeleteNamespace(t, namespace)
+	DeleteNamespace(t, traefikNamespace)
+	CheckNamespaceIsAbsent(t, namespace)
+}
+
+// Full stack scenario:
+// - Install Minio
+// - Create the Minio credentials secret
+// - Register a cluster with 3 nodes
+// - Run the Reaper test scenario
+// - Run the Medusa test scenario
+// - Run the Prometheus test scenario
+// - Run the Grafana test scenario
+// - Run the Stargate test scenario
+// - Terminate the namespace and cleanup the cluster
+func TestFullStackScenario(t *testing.T) {
+	const (
+		medusaBackend = "Minio"
+		backupName    = "backup1"
+	)
+
+	// only runs when the -short flag isn't used
+	if testing.Short() {
+		t.Skip("Skipped in short mode")
+	}
+
+	namespace := initializeCluster(t)
+	createMedusaSecretAndInstallDeps(t, namespace, medusaBackend)
+	deployFullStackCluster(t, namespace)
+
+	reaperSuccess := t.Run("Test Reaper", func(t *testing.T) {
+		testReaper(t, namespace)
+	})
+
+	medusaSuccess := t.Run("Test Medusa", func(t *testing.T) {
+		testMedusa(t, namespace, medusaBackend, backupName)
+	})
+
+	prometheusSuccess := t.Run("Test Prometheus", func(t *testing.T) {
+		testPrometheus(t, namespace)
+	})
+
+	grafanaSuccess := t.Run("Test Grafana", func(t *testing.T) {
+		testGrafana(t, namespace)
+	})
+
+	stargateSuccess := t.Run("Test Stargate", func(t *testing.T) {
+		testStargate(t, namespace)
+	})
+
+	if reaperSuccess && medusaSuccess && prometheusSuccess && grafanaSuccess && stargateSuccess {
```

Comment:
We don't, otherwise we have nothing left for investigation.
When ran locally, we want to be able to check what went wrong manually. When running in CI, in case of failure there are two steps that will trigger:

- artifacts extraction (outputs of various kubectl commands)
- enable ssh access (for manual investigation)

I don't see any drawback of not cleaning up the cluster in case of failure.

Reply:
It leaves the cluster with stuff behind it and I need to manually clean it after each test run. I shouldn't need the end results in most cases - I should be able to see the problem from the log lines and then investigate. It's a rare occurrence in my opinion that I would need to check the status that was left behind.

I'd much rather see a flag that would indicate if I want I don't want it cleaned up. I at least use the same cluster for weeks without tearing it down and I don't want to pollute it with stuff that eats memory.

Reply:
Cleanup is now done by default in all cases, unless configured otherwise. CI will not perform cleanup to allow extracting artifacts and provide ssh access for investigation.